### ISSUE TITLE
package dependency installation for subdirectories in GIT

### ISSUE THREAD

**Issue by siri16100** [r-lib/remotes#777]
I have some R packages in my companies repo. But they are not in the root path but in a subdirectory. I can easly install this packages using the `subdir` argument in `remotes::install_git`. However I do not know how so correctly specify the sub-directory in the `Remotes` section of the `DESCRIPTION` file. 

All works find if the packages are in the root directory, i.e.: 

```r
Remotes: 
  git::https://<company>@dev.azure.com/<company>/<department>/_git/<package>
``` 

but if I use something like 
```r
 Remotes: 
  git::https://<company>@dev.azure.com/<company>/<department>/_git/<reponame>/packages/<package>
``` 
(which is the url to the subdirectory \packages ) I get following error: 
```r
Error: Failed to install 'unknown package' from Git:
  can't convert package <package> with RemoteType 'git' to remote
```

**Comment by gaborcsardi** [comment]
I am afraid that there is no way to do that currently.

**Comment by kevinushey** [comment]
For what it's worth, in `renv`, we've adopted the convention of using `:` to delimit sub-directories. I added my rationale here:

https://github.com/rstudio/renv/issues/497#issuecomment-664771124

Is this something that `remotes` could also accept?

**Comment by gaborcsardi** [comment]
It is unlikely that we would implement this in remotes, but we could implement it in pak.

**Comment by siri16100** [comment]
Hi @gaborcsardi  thanks for sharing that sub-directories are currently not supported by `remotes`. As I'm not very familiar with `pak`, would a implementation of `:` as delimiter for subdirectories indeed fix the problem of specifying package dependencies in the `DESCRIPTION` file of a package?

**Comment by gaborcsardi** [comment]
I am sorry, I am not sure what you mean.

**Comment by siri16100** [comment]
Regarding my upper question you stated, that it is currently not possible to add a sub-directory GIT sources to the `Remotes` in a `DESCRIPTION` file of a package. Is there any chance this is possible in future releases? Further I don't quite get (as I'm not really familiar with `pak` )  how this:

> t is unlikely that we would implement this in remotes, but we could implement it in pak.

would help solving the issue?

**Comment by gaborcsardi** [comment]
> Is there any chance this is possible in future releases?

As we are focusing on pak, we are not adding new features to remotes, so it is unlikely.

> how this would help solving the issue?

You could use pak instead of remotes as it is mostly superior to remotes.

**Comment by siri16100** [comment]
Thanks @gaborcsardi. As I used used `renv:install` before I just played around a little with `pak:pgk_install`. I was able to install a package without any dependencies from azure. However if I try a package with dependencies I run into following error: 

 
```r
Error: 
! error in pak subprocess
Caused by error: 
! Could not solve package dependencies:
* git::https://dev.azure.com/<company>/<department>/_git/<package>:
  * Can't install dependency <package dep 1> (>= 0.1.3) (>= 0.0.3) (>= 0.0.2)
  * Can't install dependency <package dep 2> (>= 0.0.3) (>= 0.0.2)
  * Can't install dependency <package dep 3> (>= 0.0.2)
* <package dep 1>: Can't find package called <package dep 1>, <package dep 2>, <package dep 3>, <package dep 4>.
* <package dep 2>: Can't find package called <package dep 1>, <package dep 2>, <package dep 3>, <package dep 4>.
* <package dep 3>: Can't find package called <package dep 1>, <package dep 2>, <package dep 3>, <package dep 4>.
---
Backtrace:
1. pak::pkg_install("git::https://dev.azure.com/<company>/<department>/_git/<package>?reinstall")
2. pak:::remote(function(...) get("pkg_install_make_plan", asNamespace("pak"))(...), …
3. err$throw(res$error)
---
Subprocess backtrace:
1. base::withCallingHandlers(cli_message = function(msg) { …
2. get("pkg_install_make_plan", asNamespace("pak"))(...)
3. prop$stop_for_solution_error()
4. private$plan$stop_for_solve_error()
5. pkgdepends:::pkgplan_stop_for_solve_error(self, private)
6. base::throw(new_error("Could not solve package dependencies:\n", msg, …
7. | base::signalCondition(cond)
8. global (function (e) …
``` 

what I still don't get is the link between the `pak` package and this [vignette](https://remotes.r-lib.org/articles/dependencies.html). Is `pak:pgk_install`  able to make use of Remotes in a packages DESCRIPTION file? If yes I wonder why `renv:install` does work but  `pak:pgk_install` not....

**Comment by gaborcsardi** [comment]
> what I still don't get is the link between the pak package and this [vignette](https://remotes.r-lib.org/articles/dependencies.html). Is pak:pgk_install able to make use of Remotes in a packages DESCRIPTION file? 

Yes: https://pak.r-lib.org/reference/pak_package_sources.html#the-remotes-field

> If yes I wonder why renv:install does work but pak:pgk_install not....

Hard to say without seeing your `DESCRIPTION` file and knowing the layout of your git repos, unfortunately.

**Comment by gaborcsardi** [comment]
Yes, AFAICT that should work, provided that you have the credentials to access these repositories in the git credential store, and you are not running into this bug: https://github.com/r-lib/pkgdepends/issues/319

**Comment by siri16100** [comment]
Ok thanks. As I have successfully installed `pkgA` using `pak::pkg_install`, there shouldn't be any problems with the authentication I guess. But anyway, thanks for confirming that I used the right specification in `DESCRIPTION`...
